packages/babel-plugin-transform-inline-consecutive-adds/src/object-collapser.js

```diff
@@ -39,6 +39,13 @@ class ObjectCollapser extends Collapser {
       if (checkReference(right)) {
         return false;
       }
+      if (
+        obj.isIdentifier() &&
+        right.isIdentifier() &&
+        obj.node.name === right.node.name
+      ) {
```

Comment:
Good catch. But, this check should go into the `checkReference` call which is used above so it applies the fix for all collapsers.